docs/layers/tile-layer.md

```diff
@@ -126,39 +128,163 @@ Hide tiles when under-zoomed.
+##### `maxCacheSize` (Number, optional)
+
+The maximum number of tiles that can be cached. The tile cache keeps loaded tiles in memory even if they are no longer visible. It reduces the need to re-download the same data over and over again when the user pan/zooms around the map, providing a smoother experience.
 
-The maximum cache size for a tile layer. If not defined, it is calculated using the number of tiles in the current viewport times multiplied by `5`.
+If not supplied, the `maxCacheSize` is calculated as `5` times the number of tiles in the current viewport.
 
 - Default: `null`
 
 
-##### `strategy` (Enum, optional)
+##### `maxCacheByteSize` (Number, optional)
+
+The maximum memory used for caching tiles. If this limit is supplied, `getTileData` must return an object that contains a `byteLength` field.
+
+- Default: `null`
+
 
-How the tile layer determines the visibility of tiles. One of the following:
+##### `refinementStrategy` (Enum, optional)
+
+How the tile layer refines the visibility of tiles. One of the following:
 
 * `'best-available'`: If a tile in the current viewport is waiting for its data to load, use cached content from the closest zoom level to fill the empty space. This approach minimizes the visual flashing due to missing content.
 * `'no-overlap'`: Avoid showing overlapping tiles when backfilling with cached content. This is usually favorable when tiles do not have opaque backgrounds.
+* `'never'`: Do not display any tile that is not selected.
 
 - Default: `'best-available'`
 
 
-##### `tileToBoundingBox` (Function, optional)
+##### `Tileset` (Function, optional)
 
-**Advanced** Converts from `x, y, z` tile indices to a bounding box in the global coordinates. The default implementation converts an OSM tile index to `{west: <longitude>, north: <latitude>, east: <longitude>, south: <latitude>}`.
+A custom implemetation of the `Tileset2D` interface. This can be used to override some of the default behaviors of the `TileLayer`. See the `Tileset2D` documentation below.
+
+
+### Render Options
+
+##### `renderSubLayers` (Function, optional))
+
+Renders one or an array of Layer instances with all the `TileLayer` props and the following props:
+
+* `id`: An unique id for this sublayer
+* `data`: Resolved from `getTileData`
+* `tile`: An object containing tile index `x`, `y`, `z`, and `bbox` of the tile.
+
+- Default: `props => new GeoJsonLayer(props)`
 
-Receives arguments:
+
+### Callbacks
+
+##### `onViewportLoad` (Function, optional)
+
+`onViewportLoad` is a function that is called when all tiles in the current viewport are loaded. The loaded content (as returned by `getTileData`) for each visible tile is passed as an array to this callback function.
+
+- Default: `data => null`
+
+
+##### `onTileError` (Function, optional)
+
+`onTileError` called when a tile failed to load.
+
+- Default: `console.error`
+
+
+## Tileset2D
+
+> This section describes advanced customization of the TileLayer.
```

Comment:
Maybe some simpler usage exposition of the non-subclassed `Tileset2D` class before jumping into the advanced customization case?

Reply:
Going off that, is there a strong argument against baking this into deck.gl in some way? Using the non-geospatial coordinate system is something deck.gl supports but this layer as it stands is married to geo-spatial coordinates (and specifically does not have to be).  Also, is there a difference between `getTileIndicesInViewport` and `getTileIndices`?

Reply:
It should be just `getTileIndices`.

I can try integrating the non-geospatial handling. My main concern is that we don’t really have any test case for this.

Reply:
Do you mean unit testing or some sort of visual testing? I could create an image pyramid for this if you'd like.  I'd also be happy to write unit tests.

Reply:
That would be great! I'll add some unit tests before landing, but a sample set of tiles would be very helpful.